test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1347,10 +1336,19 @@ private synchronized void startAndPublishNodesAndClients(List<NodeAndClient> nod
                 // special case for 1 node master - we can't update the min master nodes before we add more nodes.
                 updateMinMasterNodes(currentMasters + newMasters);
             }
-            for (NodeAndClient nodeAndClient : nodeAndClients) {
-                nodeAndClient.startNode();
-                publishNode(nodeAndClient);
+            List<Future<?>> futures = nodeAndClients.stream().map(node -> executor.submit(node::startNode)).collect(Collectors.toList());
+            try {
+                for (Future<?> future : futures) {
+                    future.get();
+                }
+            } catch (InterruptedException e) {
+                Thread.interrupted();
+                return;
```

Comment:
maybe we can throw an AssertionError to make it clear that getting to this place is not expected (i.e. it will serve as documentation and validate our assumptions).